### ISSUE TITLE
<Button /> overriding type attribute, not submitting form

### ISSUE THREAD

**Issue by cameronjroe** [elementalui/elemental#124]
I'm trying to get a basic form going:

```
<Form onSubmit={this.handleReviewSubmission.bind(this)}>
              <FormField label="Name" htmlFor="name">
              <FormField label="Comments" htmlFor="comments">
                <FormInput 
                  name="comments" 
                  placeholder="Enter comments"  
                  multiline />
              </FormField>
              <Button type="default">Submit</Button>
</Form>
```

The button won't submit the form, also `type` is an button html attribute that should be able to be set to "submit". Am I doing something wrong?

**Comment by vdh** [comment]
After checking the source code for the Button component, it needs a `submit={true}` attribute to be a submit button. I think the documentation needs an update?

**Comment by jossmac** [comment]
Documented on http://elemental-ui.com/buttons under the first "usage" heading

**Comment by vdh** [comment]
@jossmac But the form examples on http://elemental-ui.com/forms show confusing/erroneous use of buttons labelled "Submit" that do not actually submit the form.

**Comment by jossmac** [comment]
Correct, that was an intentional decision; often I'll find myself on a UI demo site clicking around testing things and hit a button that submits with `#` dragging you back up to the top of the page.

I'll add the `submit` prop to those examples for clarity :)
